perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/impl/RegistrarManagerImpl.java

```diff
@@ -3161,6 +3156,65 @@ else if (extSourceName.equals(application.getExtSourceName()) && actor.equals(ap
 		return resultApps;
 	}
 
+
+	@Override
+	public List<Group> getGroupsForAutoRegistration(PerunSession sess, Vo vo) throws VoNotExistsException, PrivilegeException {
+		Utils.checkPerunSession(sess);
+		perun.getVosManagerBl().checkVoExists(sess, vo);
+
+		// Authorization
+		if (!AuthzResolver.authorizedInternal(sess, "getGroupsForAutoRegistration_Vo_policy", vo)) {
+			throw new PrivilegeException(sess, "getGroupsForAutoRegistration");
+		}
+
+		return perun.getGroupsManagerBl().getGroupsForAutoRegistration(sess, vo);
+	}
+
+	@Override
+	public void deleteGroupsFromAutoRegistration(PerunSession sess, List<Group> groups) throws GroupNotExistsException, PrivilegeException {
+		Utils.checkPerunSession(sess);
+
+		for(Group group : groups) {
+			perun.getGroupsManagerBl().checkGroupExists(sess, group);
+		}
+
+		//Authorization
+		for (Group group : groups) {
+			if(!AuthzResolver.authorizedInternal(sess, "deleteGroupsFromAutoRegistration_List<Group>_policy", group)) {
+				throw new PrivilegeException(sess, "deleteGroupsFromAutoRegistration");
+			}
+		}
+
+		perun.getGroupsManagerBl().deleteGroupsFromAutoRegistration(sess, groups);
+	}
+
+	@Override
+	public void addGroupsToAutoRegistration(PerunSession sess, List<Group> groups) throws GroupNotExistsException, PrivilegeException, GroupNotAllowedToAutoRegistrationException {
+		Utils.checkPerunSession(sess);
+
+		for(Group group : groups) {
+			perun.getGroupsManagerBl().checkGroupExists(sess, group);
+		}
+
+		//Authorization
+		for (Group group : groups) {
+			if(!AuthzResolver.authorizedInternal(sess, "addGroupsToAutoRegistration_List<Group>_policy", group)) {
+				throw new PrivilegeException(sess, "addGroupsToAutoRegistration");
+			}
+		}
+
+		// Create application form if non exists
+		for (Group group : groups) {
+			try {
+				getFormForGroup(group);
+			} catch (FormNotExistsException e) {
+				createApplicationFormInGroup(sess, group);
+			}
+		}
+
+		perun.getGroupsManagerBl().addGroupsToAutoRegistration(sess, groups);
```

Comment:
I like this solution since it required fewer changes to be made in the code. However, we should denote that this method `addGroupsToAutoRegistration` in the groupsManagerBl does not create the application form and that it has to be done. It should be at least mentioned in the javadoc.

Reply:
Right, I added this information to `GroupsManagerBl` javadoc.